Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -525,7 +525,13 @@ KResult VirtualFileSystem::rename(StringView old_path, StringView new_path, Cust
     if (old_parent_custody->is_readonly() || new_parent_custody->is_readonly())
         return EROFS;
 
+    auto old_basename = KLexicalPath::basename(old_path);
+    if (old_basename.is_whitespace() || old_basename == "."sv || old_basename == ".."sv)
```

Comment:
Does Dr.POSIX require this? AFAIK whitespace is a perfectly valid file name.

Reply:
Indeed, whitespace filenames are valid:
```console
$ touch "   "
$ ls
'   '
```

Reply:
Thank you, I rephrased the commit messages and changed the check to only disallow empty filenames.